File: questions/Q1.py
def insert(a, n):
    for i in range(1, n):
        temp = a[i]
        j = i - 1
        while j >= 0:
            a[j + 1] = a[j]
            j -= 1
        a[j + 1] = temp

def print_arr(a, n):
    for i in range(n):
        print(f"{a[i]} ", end='')

if __name__ == "__main__":
    a = [12, 31, 25, 8, 32, 17]
    n = len(a)

    print("Before sorting array elements are - ")
    print_arr(a, n)

    insert(a, n)

    print("\nAfter sorting array elements are - ")
    print_arr(a, n)

File: set2/Q1.c
#include <stdio.h>

void Sort(int arr[], int n) {  
  int i, j;
  for (i = 0; i <= n - 1; i++) {
    for (j = 0; j < n - i - 1; j += 2) {
      arr[j] = arr[j] + arr[j + 1];
      arr[j + 1] = arr[j] - arr[j + 1];
      arr[j] = arr[j] - arr[j + 1];
    }
  }
}

void printArr(int a[], int n) {
  int i;
  for (i = 0; i < n; i++)
    printf("%d ", a[i]);
}

int main() {
  int arr[] = {64, 34, 25, 12, 22, 11, 90};
  int n = sizeof(arr) / sizeof(arr[0]);

  printf("Before sorting array elements are - \n");
  printArr(arr, n);

  Sort(arr, n);

  printf("\nAfter sorting (potentially incorrect) array elements are - \n");
  printArr(arr, n);

  return 0;
}
